doc/fiber.md

```diff
@@ -1,16 +1,46 @@
-# Scheduler
+# Fiber
+
+Fibers provide a mechanism for cooperative concurrency.
+
+## Context Switching
+
+Fibers execute a user provided block. During the execution, the block may call `Fiber.yield` or `Fiber.transfer` to switch to another fiber. `Fiber#resume` is used to continue execution from the point where `Fiber.yield` was called.
+
+``` ruby
+#!/usr/bin/env ruby
+
+puts "1: Start program."
+
+f = Fiber.new do
+  puts "3: Entered fiber."
+  Fiber.yield
+  puts "5: Resumed fiber."
+end
+
+puts "2: Resume fiber first time."
+f.resume
+
+puts "4: Resume fiber second time."
+f.resume
+
+puts "6: Finished."
+```
+
+This program demonstrates the flow control of fibers.
+
+## Scheduler
```

Comment:
don't you add about the following?

* `Fiber.set_scheduler`
* `Fiber.new(blocking: true)`

If there is callback specification, it will be helpful.

Reply:
I'll check where it makes sense to add documentation, maybe makes more sense in `cont.c`. Thanks for bringing it up.

Reply:
Okay, I added it to `fiber.md` but we might also need to add it to the `cont.c` documentation too.